Refact Remote Kernel Connection UI

samborba opened the pull request:
- rename all "local" to "remote", since the context is in PlatIAgro's perspective
- remove remote connection button icon
- add visual connection and disconnection feedback

samborba committed:
Refact Remote Kernel Connection UI
- rename all "local" to "remote", since the context is in PlatIAgro's perspective
- remove remote connection button icon
- add visual connection and disconnection feedback

samborba committed:
Rename Tracking Variable
- Use `currentKernel` instead `previousKernel`

samborba committed:
Improve UI Feedbacks
- add class name in the toolbar items
- update remote kernel button when connecting to a kernel
- better handle socket errors
- refact .png type

fberanizo reviewed (commented):
@samborba pode me explicar o que cada um dos status abaixo significa?
![image](https://user-images.githubusercontent.com/1791381/100252582-cd9bdb80-2f1e-11eb-913a-45ce5a993299.png)

![image](https://user-images.githubusercontent.com/1791381/100252863-253a4700-2f1f-11eb-88ad-e0c47940efdd.png)


![image](https://user-images.githubusercontent.com/1791381/100252647-db516100-2f1e-11eb-9d1f-22d91d1b3d05.png)

![image](https://user-images.githubusercontent.com/1791381/100252686-e99f7d00-2f1e-11eb-8f6a-6b48285a82af.png)

---
## Se eu me conectei em um kernel remoto e quiser voltar para a platia, tem como?

samborba commented:
> @samborba pode me explicar o que cada um dos status abaixo significa?
> ![image](https://user-images.githubusercontent.com/1791381/100252582-cd9bdb80-2f1e-11eb-913a-45ce5a993299.png)

Significa que você está conectado ao kernel remoto

> ![image](https://user-images.githubusercontent.com/1791381/100252863-253a4700-2f1f-11eb-88ad-e0c47940efdd.png)

Possível bug

> ![image](https://user-images.githubusercontent.com/1791381/100252647-db516100-2f1e-11eb-9d1f-22d91d1b3d05.png)

Bug

> ![image](https://user-images.githubusercontent.com/1791381/100252686-e99f7d00-2f1e-11eb-8f6a-6b48285a82af.png)

Estado inicial da extensão

> ## Se eu me conectei em um kernel remoto e quiser voltar para a platia, tem como?
Sim. Ao se conectar no kernel remoto é exibido um texto dentro do dialog que explica como se desconectar: basta selecionar um outro kernel (botão nativo do Jupyter). Caso ache que não esteja muito claro essa parte, eu posso esconder esse botão de conectar ao kernel remoto e exibir um para se desconectar usando o DOM. O que acha?

Você consegue me informar o passo a passo que te levou para o resultado da imagem 2?

fberanizo commented:
> Você consegue me informar o passo a passo que te levou para o resultado da imagem 2?

1o me conectei no kernel remoto (minha máquina), depois fui no menu kernel -> shutdown kernel
Eu estava tentando reconectar na platia...

O que achei que ficou estranho:
- Antes de conectar remoto o Botão é `Remote Kernel Connection`
  Depois de conectar remoto às vezes fica  `(ícone-check) Remote Kernel Connection`  outras vezes `Connected (local)`. 
Poderia ser `Connected (remote)` acho... Mas não entendi pq cada hora fica de um jeito.

samborba committed:
Fix Bug
- remove check icon
- add namespace for private data

samborba commented:
> > Você consegue me informar o passo a passo que te levou para o resultado da imagem 2?
> 
> 1o me conectei no kernel remoto (minha máquina), depois fui no menu kernel -> shutdown kernel
> Eu estava tentando reconectar na platia...
> 
> O que achei que ficou estranho:
> 
>     * Antes de conectar remoto o Botão é `Remote Kernel Connection`
>       Depois de conectar remoto às vezes fica  `(ícone-check) Remote Kernel Connection`  outras vezes `Connected (local)`.
>       Poderia ser `Connected (remote)` acho... Mas não entendi pq cada hora fica de um jeito.

@fberanizo depois do seu feedback, melhorei a parte de se desconectar do kernel remoto, porém encontrei algumas dificuldades em deixar o título do botão dinâmico, então decidi colocar um `tooltip` mais genérico:
![tooltip](https://user-images.githubusercontent.com/56934233/100367080-79593000-2fe0-11eb-90f8-9a1bc7844e6a.png)

Para se desconectar do kernel remoto é só pressionar novamente o mesmo botão e o seguinte dialog irá aparecer:
![disconect-dialog](https://user-images.githubusercontent.com/56934233/100366966-4dd64580-2fe0-11eb-8f8c-88ae82d8d642.png)

O título do botão ainda não está muito genérico. Você teria alguma sugestão de título?

samborba committed:
Handle No Kernel Status
- fix a bug that occurs when there was no kernel available in the session and is requested by a remote connection

fberanizo reviewed (approved):
💯 :)
